artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/Page.java

```diff
@@ -312,6 +451,10 @@ public void close() throws Exception {
     * While reading the cache we don't need (and shouldn't inform the backup
     */
    public synchronized void close(boolean sendEvent) throws Exception {
+      if (readFileBuffer != null) {
+         fileFactory.releaseBuffer(readFileBuffer);
```

Comment:
Currently the NIO pool is thread local, so I'm not sure is a good idea to do this on different threads (very likely).
We could implement a non-thread local pool or (better), do not use the pool but directly direct ByteBuffer instances obtained from the factory